src/app/authentication/authentication.d.ts

```diff
@@ -0,0 +1,28 @@
+export interface AuthenticationState {
+    isFetching: boolean;
+    error?: Error;
+    username?: string;
+    token?: string;
+}
+
+export interface LoginPayload {
+    username: string;
+    password: string;
+}
+
+export interface RegisterPayload {
+    username: string;
+    password: string;
+    login: string;
+}
+
+export interface AuthCompletedPayload {
+    username: string;
+    token: string;
+}
+
+export interface RegisterCompletedPayload {
+    username: string;
+    login: string;
+    password: string;
```

Comment:
>username
>>login
>>>password

x3 lul

Reply:
BTW why we need this type?

Reply:
because when we register we don't need token I guess